### PR TITLE
tsp, rename folder

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
acr changed again, regening tests. https://github.com/Azure/azure-sdk-for-java/pull/34070

### CODE REVIEW

File: eng/pipelines/ci-cadl-java.yaml [hunk 1]
```diff
@@ -71,14 +71,14 @@ jobs:
       - script: |
           node node_modules/\@azure-tools/cadl-ranch/dist/cli/cli.js serve node_modules/\@azure-tools/cadl-ranch-specs/http/ --coverageFile ./cadl-ranch-coverage-java.json &
```
Comment (haolingdong-msft) [inline eng/pipelines/ci-cadl-java.yaml:72]: do we want do change the file name to ci-typespec-java.yaml?

File: typespec-extension/src/main/java/com/azure/cadl/Main.java [hunk 2]
```diff
@@ -50,11 +50,11 @@ public class Main {
                 .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
     }
```
Comment (haolingdong-msft) [inline typespec-extension/src/main/java/com/azure/cadl/Main.java:51]: Do we want to change the namespace to `package com.azure.typespec;`?

**Review by haolingdong-msft — APPROVED** [review]

File: typespec-extension/src/main/java/com/azure/cadl/Main.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply typespec-extension/src/main/java/com/azure/cadl/Main.java:51]: java namespace currently is not exposed anywhere, so no change for now (include the name of `CadlPlugin`)

File: eng/pipelines/ci-cadl-java.yaml (hunk 1 above)
Comment (weidongxu-microsoft) [reply eng/pipelines/ci-cadl-java.yaml:72]: renamed, hope CI still works

**Pull request merged**